Dockerfile

```diff
@@ -39,13 +45,30 @@ ENV NEXTAUTH_SECRET=$NEXTAUTH_SECRET
 ENV NEXTAUTH_URL=$NEXTAUTH_URL
 ENV PORT 3000
 
-COPY ./ ./
+FROM base AS builder
+WORKDIR /src
+COPY --from=deps /src/node_modules ./node_modules
+COPY --from=deps /src/app/node_modules ./app/node_modules
+COPY . .
 
 RUN yarn prisma generate
 RUN yarn build
 
-# Install only prod dependencies and start app
-RUN yarn install --frozen-lockfile --production && yarn cache clean
-CMD npm start
+FROM base AS runner
+WORKDIR /src
+
+RUN addgroup --system --gid 1001 nodejs
+RUN adduser --system --uid 1001 nextjs
+
+COPY --from=builder /src/app/public ./app/public
+
+COPY --from=builder /src/app/.next/standalone ./
+COPY --from=builder /src/app/.next/static ./app/.next/static
+
+USER nextjs
 
 EXPOSE 3000
+
+ENV PORT 3000
+
+CMD HOSTNAME="0.0.0.0" node app/server.js
```

Comment:
Why not npm start ?

Reply:
Might be the case, I adapted the official Next.js Dockerfile example from [here](https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile). Let me know if you see some issues with this approach 👀

Reply:
Here it seems we would miss the db:migrate script and potentially some environment variables (although I think I am not sure GLOBAL_AGENT and NO_PROXY are needed in case of visualize; I know they are in Elcom though), and the $PORT ? Thanks for the link to the doc, I know understand what is this app/server.js, I think we should have a docker:start command in npm scripts, so that we can handle additional scripts and make sure "start" and "docker:start" are aligned.

Reply:
Ah yes, I see, thanks for pointing this out! I thought we kind of have an equivalent of db:migrate script in the Dockerfile, when we `RUN yarn prisma generate`, but this might not be the case. I'll try to add the docker:start command and align both 👍

Reply:
@ptbrowne done in https://github.com/visualize-admin/visualization-tool/pull/1445/commits/3527df544d2ea0ee86722acef2237bec24ba96d6. Not sure if we need to be able to control $PORT as we also exposed port 3000 in the old Docker file? Or did I miss something? 😄

Reply:
Yep, not sure about that, let's try without port then ;)